### PR TITLE
Created StateTreeManager scaffolding

### LINKED ISSUE #176: BeaconState is uninitialized at the start of the BeaconChainService
Modify BeaconChainService to initialize the state of the BeaconState and BeaconBlock.

### LINKED ISSUE #194: Create a StateTreeManager.java
### Description
At each slot, if there is a new block to process, or a new attestation received, then we need to rescore all potential branches that could take the place of the branch that represents the current slots state. If one of those branches scores higher, then it will become the current state at the current slots.  We need a StateTreeManager to handle all the state switching and calls to the BlockStoreService.

### Acceptance Criteria
* Initial scaffolding - stubbed methods

### PR DESCRIPTION
**Pull request by jrhea**
## PR Description
Created initial StateTree Manager.  Moved slot, block and attestation events to the StateTreeManager class.  Also added dummy blocks and attestations to keep the StateTransition logic from failing.
## Fixed Issue(s)

fixes #194 
fixes #176

### CODE REVIEW

File: networking/p2p/build.gradle [hunk 1]
```diff
@@ -7,7 +7,9 @@ jar {
 }
 
 dependencies {
-
+   implementation project(':util')
```
Comment (schroedingerscode) [inline networking/p2p/build.gradle]: Is there an extra space here or is it just rendering wrong?

File: ethereum/beaconchain/src/main/java/tech/pegasys/artemis/state/StateTreeManager.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 ConsenSys AG.
```
Comment (schroedingerscode) [inline ethereum/beaconchain/src/main/java/tech/pegasys/artemis/state/StateTreeManager.java]: 2019 on the new file.

File: networking/p2p/build.gradle (hunk 1 above)
Comment (akhila-raju) [reply networking/p2p/build.gradle]: The extra space was removed

**Review by akhila-raju — APPROVED** [review]

**Review by akhila-raju — CHANGES_REQUESTED** [review]
LGTM after changes made to Steven's comment.

File: networking/p2p/build.gradle (hunk 1 above)
Comment (schroedingerscode) [reply networking/p2p/build.gradle]: Sorry, confusing terminology. Newline was removed but it looks like there's 3 spaces indenting Line 10, whereas there's 2 on each line under it.

File: ethereum/beaconchain/src/main/java/tech/pegasys/artemis/state/StateTreeManager.java (hunk 2 above)
Comment (jrhea) [reply ethereum/beaconchain/src/main/java/tech/pegasys/artemis/state/StateTreeManager.java]: How do u notice this stuff?

File: networking/p2p/build.gradle (hunk 1 above)
Comment (jrhea) [reply networking/p2p/build.gradle]: you aren't human

**Review by schroedingerscode — APPROVED** [review]
LGTM

File: networking/p2p/build.gradle (hunk 1 above)
Comment (akhila-raju) [reply networking/p2p/build.gradle]: LOL

**Review by akhila-raju — APPROVED** [review]

**Pull request merged**